javascript/react/src/hooks/columns/useKeyNames.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import type ColumnTable from 'arquero/dist/types/table/column-table'
+import { useMemo } from 'react'
+import type { RowObject } from 'arquero/dist/types/table/table'
+
+export function useKeyNames(
+	table: ColumnTable | undefined,
+	column: string,
+): string[] {
+	return useMemo(() => {
+		const distinctKeys = new Set<string>()
+
+		if (table !== undefined) {
+			const columnSelected: RowObject[] = table.select(column).objects()
```

Comment:
I would try to use native arquero functions as much as possible before switching to objects. For example, the op.keys can get the keys into an array https://uwdata.github.io/arquero/api/op#keys. I think you could then do an unroll verb, and finally and array_agg_distinct to get the unique list of keys for all rows. 

I’m not sure if this is faster, but usually arquero does a bunch of optimization internally that makes it much better than switching to the raw objects.

Reply:
I was able to replace the logic using op.keys but the op. array_agg_distinct did not work for me (you can see the updates on the PR)